### PR TITLE
Add a frequency to allow logging less often

### PR DESCRIPTION
**Pull request by jlogeart**

### CODE REVIEW

File: src/main/scala/io/flow/log/RollbarLogger.scala [hunk 1]
```diff
@@ -99,30 +108,35 @@ case class RollbarLogger @AssistedInject() (
...
 
-  def error(message: => String, error: => Throwable): Unit = {
-    if (logger.isErrorEnabled)
-      logger.error(appendEntries(convert(attributes)), legacyMessage.getOrElse(message), error)
-    if (shouldSendToRollbar)
-      rollbar.foreach(_.error(error, convert(attributes), message))
-  }
+  def error(message: => String, error: => Throwable): Unit =
+    if (shouldLog) {
+      if (shouldLog && logger.isErrorEnabled)
```
Comment (benwaffle) [inline src/main/scala/io/flow/log/RollbarLogger.scala:133]: redundant shouldLog check here

File: src/main/scala/io/flow/log/LogUtil.scala [hunk 2]
```diff
@@ -82,4 +82,6 @@ class LogUtil @Inject() (logger: RollbarLogger) {
       f
   }
 
+  def shouldLog(frequency: Long): Boolean = frequency == 1L || Random.nextLong() % frequency == 0
```
Comment (benwaffle) [inline src/main/scala/io/flow/log/LogUtil.scala:85]: is this used anywhere?

File: src/main/scala/io/flow/log/RollbarLogger.scala [hunk 3]
```diff
@@ -99,30 +108,35 @@ case class RollbarLogger @AssistedInject() (
...
   def error(message: => String): Unit = error(message, null)
 
-  def debug(message: => String, error: => Throwable): Unit = {
-    if (logger.isDebugEnabled)
+  def debug(message: => String, error: => Throwable): Unit =
+    if (shouldLog && logger.isDebugEnabled) {
       logger.debug(appendEntries(convert(attributes)), legacyMessage.getOrElse(message), error)
-    //not sending to rollbar to save quota
-  }
+      // not rollbar to save quota
```
Comment (benwaffle) [inline src/main/scala/io/flow/log/RollbarLogger.scala:114]: can you revert to the old comment?

File: src/main/scala/io/flow/log/RollbarLogger.scala [hunk 4]
```diff
@@ -99,30 +108,35 @@ case class RollbarLogger @AssistedInject() (
...
+    }
 
-  def info(message: => String, error: => Throwable): Unit = {
-    if (logger.isInfoEnabled)
+  def info(message: => String, error: => Throwable): Unit =
+    if (shouldLog && logger.isInfoEnabled) {
       logger.info(appendEntries(convert(attributes)), legacyMessage.getOrElse(message), error)
-    //not sending to rollbar to save quota
-  }
+      // not rollbar to save quota
```
Comment (benwaffle) [inline src/main/scala/io/flow/log/RollbarLogger.scala:120]: can you revert to the old comment?

File: src/main/scala/io/flow/log/LogUtil.scala (hunk 2 above)
Comment (jlogeart) [reply src/main/scala/io/flow/log/LogUtil.scala:85]: Not sure what you asking for, since this is new functionality

Comment (jlogeart) [reply src/main/scala/io/flow/log/LogUtil.scala:85]: My mistake - pushed too early

**Pull request merged**